File: examples/csharp/avalonia/NotesShowcase/Models/NotebookModel.cs
namespace NotesShowcase.Models;

/// <summary>
/// Domain model for a notebook (a node in the notebooks tree).
///
/// See spec/proposals/2026-05-29-notes-showcase-scenario.md §3.1.
/// Pure data — no behavior, no VMx dependencies.
/// </summary>
public sealed record NotebookModel(string Id, string Name, string? ParentId);

File: examples/swift/notes-showcase/Sources/NotesShowcaseCore/Models/NotebookModel.swift
import Foundation

/// Domain model for a notebook (a node in the notebooks tree).
///
/// See spec/proposals/2026-05-29-notes-showcase-scenario.md §5.3.
/// Pure data — no behavior, no VMx dependencies.
public struct NotebookModel: Sendable, Equatable, Codable {
    public let id: String
    public let name: String
    public let parentId: String?
    public let isReadonly: Bool

    public init(id: String, name: String, parentId: String?, isReadonly: Bool = false) {
        self.id = id
        self.name = name
        self.parentId = parentId
        self.isReadonly = isReadonly
    }
}
